tutordeck/server/templates/plugin.html

```diff
@@ -43,66 +44,72 @@ <h3>Default settings</h3>
-            pluginUpgradeButton.style.display = 'none'; 
-            cancelCommandButton.style.display = 'block'; 
-        }
-        function ShowRunCommandButton(){
-            if (isPluginInstalled){
-                showPluginUpgradeButton();
-            } else {
-                showPluginInstallButton();
-            }
-        }
-        function showPluginEnableDisableBar() {
-            const bar = document.getElementById('plugin-enable-disable-bar');
-            bar.style.display = isPluginInstalled === true ? 'flex' : 'none';
+    function showPluginInstallButton() {
+        pluginInstallButton.style.display = 'block';
+        pluginUpgradeButton.style.display = 'none';
+        cancelCommandButton.style.display = 'none';
+    }
+    function showPluginUpgradeButton() {
+        pluginInstallButton.style.display = 'none';
+        pluginUpgradeButton.style.display = 'block';
+        cancelCommandButton.style.display = 'none';
+    }
+    function ShowCancelCommandButton() {
+        pluginInstallButton.style.display = 'none';
+        pluginUpgradeButton.style.display = 'none';
+        cancelCommandButton.style.display = 'block';
+    }
+    function ShowRunCommandButton() {
+        if (isPluginInstalled) {
+            showPluginUpgradeButton();
+        } else {
+            showPluginInstallButton();
         }
+    }
+    function showPluginEnableDisableBar() {
+        const bar = document.getElementById('plugin-enable-disable-bar');
+        bar.style.display = isPluginInstalled === true ? 'flex' : 'none';
+    }
+    async function checkIfPluginInstalled(pluginName) {
+        const response = await fetch(`/plugin/${pluginName}/is-installed`);
+        const data = await response.json();
+        return data.installed;
+    }
 
-        showPluginEnableDisableBar();
-	    ShowRunCommandButton();
+    showPluginEnableDisableBar();
+    ShowRunCommandButton();
 
-        // Add change event to all inputs, selects
-        document.querySelectorAll('#config-forms-container input').forEach(function(element) {
-            element.addEventListener('change', function() {
-                this.classList.add('changed');
-                // Find the associated hidden input
-                const hiddenInput = this.nextElementSibling;
-                if (hiddenInput && hiddenInput.type === 'hidden') {
-                    hiddenInput.classList.add('changed');
-                }
-            });
+    // Add change event to all inputs, selects
+    document.querySelectorAll('#config-forms-container input').forEach(function (element) {
+        element.addEventListener('change', function () {
+            this.classList.add('changed');
+            // Find the associated hidden input
+            const hiddenInput = this.nextElementSibling;
+            if (hiddenInput && hiddenInput.type === 'hidden') {
+                hiddenInput.classList.add('changed');
+            }
         });
```

Comment:
It's always a good idea to use arrow functions over functions defined with the 'function' keyword, especially when using the 'this' keyword. 

```suggestion
    document.querySelectorAll('#config-forms-container input').forEach((element) => {
        element.addEventListener('change', () => {
            element.classList.add('changed');
            // Find the associated hidden input
            const hiddenInput = element.nextElementSibling; // 'this' won't work, so using element.
            if (hiddenInput && hiddenInput.type === 'hidden') {
                hiddenInput.classList.add('changed');
            }
        });
```